### PR TITLE
feature/feedback

### PR DESCRIPTION
**Pull request by AzamatMuratbekov**

### CODE REVIEW

File: src/main/java/us/peaksoft/gadgetarium/controller/FeedbackController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.web.bind.annotation.*;
+import us.peaksoft.gadgetarium.dto.FeedbackRequest;
+import us.peaksoft.gadgetarium.dto.FeedbackResponce;
+import us.peaksoft.gadgetarium.dto.RatingResponce;
+import us.peaksoft.gadgetarium.service.FeedbackService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/public/feedback")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/FeedbackController.java:13]: pls remove public

File: src/main/java/us/peaksoft/gadgetarium/controller/FeedbackController.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return feedbackService.getById(id);
+    }
+
+    @PutMapping("{id}")
+    public FeedbackResponce update(@PathVariable("id") Long id, @RequestBody FeedbackRequest feedbackRequest) {
+        return feedbackService.update(id, feedbackRequest);
+    }
+
+    @DeleteMapping("{id}")
+    public String delete(@PathVariable("id") Long id) {
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/FeedbackController.java:40]: Use SimpleResponse instead String

File: src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Bean
+    public AuthenticationProvider authenticationProvider(){
+        DaoAuthenticationProvider authProvider = new DaoAuthenticationProvider();
+        authProvider.setUserDetailsService(userDetailsService());
+        authProvider.setPasswordEncoder(passwordEncoder());
+        return authProvider;
+    }
+    @Bean
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java:34]: Write in the next line

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class SecurityConfig {
+    private final AuthenticationProvider authenticationProvider;
+    @Bean
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:17]: Next line

**Review by i-Coderr — COMMENTED** [review]
Commented at 9:50 01/04/2023

File: src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java (hunk 3 above)
Comment (AzamatMuratbekov) [reply src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java:34]: There are in master

File: src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java (hunk 4 above)
Comment (AzamatMuratbekov) [reply src/main/java/us/peaksoft/gadgetarium/security/SecurityConfig.java:17]: done

File: src/main/java/us/peaksoft/gadgetarium/controller/FeedbackController.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+public class FeedbackController {
+    private final FeedbackService feedbackService;
+
+    @GetMapping
+    public List<FeedbackResponce> getAllFeedbacks() {
+        return feedbackService.getAllFeedbacks();
+    }
+
+    @PostMapping
+    public String save(@RequestBody FeedbackRequest feedbackRequest) {
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/FeedbackController.java:25]: Return SimpleResponse

**Review by Aidai0404 — COMMENTED** [review]
done!

**Pull request closed without merging**